### PR TITLE
SALTO-2349: Issue layout fix

### PR DESCRIPTION
**Pull request by idoamit8**
This PR follows [support RequestType and IssueView](https://github.com/salto-io/salto/pull/4878). 
I only change the arrangement and the arguemant of the function so it will support more issue layouts. 

---

_Additional context for reviewer_

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/jira-adapter/src/filters/layouts/layout_types.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export const createIssueLayoutType = (): {
   })
 
   const issueLayoutType = new ObjectType({
```
Comment (IdoZakk) [inline packages/jira-adapter/src/filters/layouts/layout_types.ts:42]: nit: search and replace issueLayout to layout in the entire folder

File: packages/jira-adapter/src/filters/layouts/layouts_utils.ts [hunk 2]
```diff
@@ -0,0 +1,197 @@
+/*
```
Comment (IdoZakk) [inline packages/jira-adapter/src/filters/layouts/layouts_utils.ts:1]: This is a major layout file, the name utils is not a perfect match
I would split to `layout_deploy` and `layout_fetch`, or change to something like `layout_service_operations`

File: packages/jira-adapter/src/filters/layouts/layouts_utils.ts [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    return response
+  } catch (e) {
+    log.error(`Failed to get issue layout for project ${variables.projectId} and screen ${variables.extraDefinerId}: ${e}`)
+  }
+  return { data: undefined }
+}
+
+const fromIssueLayoutConfigRespToIssueLayoutConfig = (
+  containers: containerIssueLayoutResponse[]
+):
```
Comment (IdoZakk) [inline packages/jira-adapter/src/filters/layouts/layouts_utils.ts:75]: Supernit: no reason for a new line

File: packages/jira-adapter/src/filters/sort_lists.ts [hunk 4]
```diff
@@ -81,9 +81,6 @@ const VALUES_TO_SORT: Record<string, Record<string, string[]>> = {
   [WORKFLOW_STATUS_TYPE_NAME]: {
     properties: ['key'],
   },
-  [ISSUE_LAYOUT_TYPE]: {
-    owners: ['elemID.name'],
-  },
```
Comment (IdoZakk) [inline packages/jira-adapter/src/filters/sort_lists.ts:86]: Why is it out? no need to sort them anymore?

File: packages/jira-adapter/src/reference_mapping.ts [hunk 5]
```diff
@@ -911,12 +911,6 @@ export const referencesRules: JiraFieldReferenceDefinition[] = [
     jiraMissingRefStrategy: 'typeAndValue',
     target: { type: 'Screen' },
   },
-  {
-    src: { field: 'owners', parentTypes: [ISSUE_LAYOUT_TYPE] },
-    serializationStrategy: 'id',
-    jiraMissingRefStrategy: 'typeAndValue',
-    target: { type: ISSUE_TYPE_NAME },
-  },
```
Comment (IdoZakk) [inline packages/jira-adapter/src/reference_mapping.ts:919]: Why is this out?

**Review by IdoZakk — COMMENTED** [review]
🏃‍♂️ 🏆 
There were two changes I did not understand there, and a file better split

File: packages/jira-adapter/src/filters/sort_lists.ts (hunk 4 above)
Comment (idoamit8) [reply packages/jira-adapter/src/filters/sort_lists.ts:86]: we are not fetching them at all, so no reason to keep it

File: packages/jira-adapter/src/reference_mapping.ts (hunk 5 above)
Comment (idoamit8) [reply packages/jira-adapter/src/reference_mapping.ts:919]: we are not fetching owners field in issue layout

File: packages/jira-adapter/src/filters/sort_lists.ts (hunk 4 above)
Comment (IdoZakk) [reply packages/jira-adapter/src/filters/sort_lists.ts:86]: Next time please write about those unrelated type of changes in the PR comments

File: packages/jira-adapter/src/filters/layouts/issue_layout.ts [hunk 6]
```diff
@@ -66,7 +66,8 @@ const filter: FilterCreator = ({ client, config, fetchQuery, getElemIdFunc }) =>
     )
     const projectToScreenId = Object.fromEntries(Object.entries(await getProjectToScreenMapping(elements))
       .filter(([key]) => Object.keys(projectIdToProject).includes(key)))
-    const { issueLayoutType, subTypes } = createLayoutType(ISSUE_LAYOUT_TYPE)
+    const { subTypes } = createLayoutType(ISSUE_LAYOUT_TYPE)
+    const issueLayoutType = createLayoutType(ISSUE_LAYOUT_TYPE).layoutType
```
Comment (IdoZakk) [inline packages/jira-adapter/src/filters/layouts/issue_layout.ts:70]: You can do 
    const { subTypes, layoutType: issueLayoutType } = createLayoutType(ISSUE_LAYOUT_TYPE)

**Review by IdoZakk — APPROVED** [review]
💯 
See comments for a small nit

**Pull request merged**